sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/Transforms.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics;
+
+import com.azure.ai.textanalytics.implementation.models.DocumentStatistics;
+import com.azure.ai.textanalytics.implementation.models.MultiLanguageInput;
+import com.azure.ai.textanalytics.implementation.models.RequestStatistics;
+import com.azure.ai.textanalytics.implementation.models.TextAnalyticsError;
+import com.azure.ai.textanalytics.models.ErrorCodeValue;
+import com.azure.ai.textanalytics.models.TextDocumentBatchStatistics;
+import com.azure.ai.textanalytics.models.TextDocumentInput;
+import com.azure.ai.textanalytics.models.TextDocumentStatistics;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Helper class to convert service level models to SDK exposes models.
+ */
+class Transforms {
+
+    /**
+     * Given a list of inputs will apply the indexing function to it and return the updated list.
+     *
+     * @param textInputs the inputs to apply the mapping function to.
+     * @param mappingFunction the function which applies the index to the incoming input value.
+     * @param <T> the type of items being returned in the list.
+     * @return The list holding all the generic items combined.
+     */
+    static <T> List<T> mapByIndex(List<String> textInputs, BiFunction<String, String, T> mappingFunction) {
+        return IntStream.range(0, textInputs.size())
+            .mapToObj(index -> mappingFunction.apply(String.valueOf(index), textInputs.get(index)))
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Convert {@link DocumentStatistics} to {@link TextDocumentStatistics}
+     *
+     * @param statistics the {@link DocumentStatistics} provided by the service.
+     * @return the {@link TextDocumentStatistics} returned by the SDK.
+     */
+    static TextDocumentStatistics toTextDocumentStatistics(DocumentStatistics statistics) {
+        return new TextDocumentStatistics(statistics.getCharactersCount(), statistics.getTransactionsCount());
+    }
+
+    /**
+     * Convert {@link RequestStatistics} to {@link TextDocumentBatchStatistics}
+     *
+     * @param statistics the {@link RequestStatistics} provided by the service.
+     * @return the {@link TextDocumentBatchStatistics} returned by the SDK.
+     */
+    static TextDocumentBatchStatistics toBatchStatistics(RequestStatistics statistics) {
+        return new TextDocumentBatchStatistics(statistics.getDocumentsCount(), statistics.getErroneousDocumentsCount(),
```

Comment:
since you are refactoring 
switch ` statistics.getErroneousDocumentsCount()` and  `statistics.getValidDocumentsCount()` will resolve this issue #7072 I found while writing samples.